app/routes.py

```diff
@@ -1 +1,191 @@
+def delete_one_task(task_id):
+    task = validate_item(Task, task_id)
+    
+    db.session.delete(task)
+    db.session.commit()
+
+    return {"details": f'Task {task_id} "{task.title}" successfully deleted'}, 200
+
+#ROUTE WITH SLACK API CALL
+@task_bp.route("/<task_id>/mark_complete", methods = ["PATCH"])
+def mark_task_complete(task_id):
+    updated_task = validate_item(Task, task_id)
+
+    updated_task.completed_at = datetime.now()
+    db.session.commit()
+    
+    channel_id = "C056SJEC9D0"
+    try:
+        # Call the chat.postMessage method using the WebClient
+        result = client.chat_postMessage(
+            channel=channel_id, 
+            text=f"Someone just completed the task {updated_task.title}"
+        )
+        logger.info(result)
+
+    except SlackApiError as e:
+        logger.error(f"Error posting message: {e}")
+    return {"task": updated_task.to_dict()}, 200
+
+@task_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"])
+def mark_task_incomplete(task_id):
+    updated_task = validate_item(Task, task_id)
+
+    updated_task.completed_at = None
+    db.session.commit()
+    return {"task": updated_task.to_dict()}, 200
+
+#GOAL ROUTES
+@goal_bp.route("", methods = ["POST"])
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(
+            title = request_body["title"]
+        )
+        db.session.add(new_goal)
+        db.session.commit()
+
+        result = new_goal.to_dict()
+        return {"goal": result}, 201
+    except:
+        return abort(make_response({"details": "Invalid data"}, 400))
+    
+@goal_bp.route("", methods = ["GET"])
+def get_goals():
+    response = []
+    all_goals = Goal.query.all()
+    for goal in all_goals:
+        response.append(goal.to_dict())
+    return jsonify(response), 200
+
+@goal_bp.route("/<goal_id>", methods = ["GET"])
+def get_goal_by_id(goal_id):
+    goal = validate_item(Goal, goal_id)
+    return {"goal": goal.to_dict()}, 200
+
+@goal_bp.route("/<goal_id>", methods = ["PUT"])
+def update_one_goal(goal_id):
+    updated_goal = validate_item(Goal, goal_id)
+
+    request_body = request.get_json()
+    updated_goal.title = request_body["title"]
+
+    db.session.commit()
+    return {"goal": updated_goal.to_dict()}, 200
+
+@goal_bp.route("/<goal_id>", methods = ["DELETE"])
+def delete_one_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+    
+    db.session.delete(goal)
+    db.session.commit()
+
+    return {"details": f'Goal {goal_id} "{goal.title}" successfully deleted'}, 200
+
+#NESTED ROUTES
+@goal_bp.route("/<goal_id>/tasks", methods= ["POST"])
+def post_tasks_under_goal(goal_id):
+    goal = validate_item(Goal, goal_id)
+
+    request_body = request.get_json()
+    try:
+        new_tasks_for_goal = request_body["task_ids"]
+        tasks = []
+        for task_id in new_tasks_for_goal:
+            tasks.append(validate_item(Task, task_id))
+
+        goal.tasks = tasks
```

Comment:
If the goal already had tasks attached to it, this would get rid of them. It would be better to append the tasks.